lib/spack/spack/build_systems/cmake.py

```diff
@@ -185,13 +184,9 @@ def _std_args(pkg):
             define('CMAKE_INSTALL_RPATH_USE_LINK_PATH', False),
             define('CMAKE_INSTALL_RPATH',
                    spack.build_environment.get_rpaths(pkg)),
+            define('CMAKE_PREFIX_PATH',
+                   spack.build_environment.get_cmake_prefix_path(pkg))
```

Comment:
Why not `os.environ.get('CMAKE_PREFIX_PATH')`? We know we set it in build_environment.

Reply:
I found that too opaque... also not really easier to write tests for

Reply:
Another question: why define it both in the environment and on the command line? Why not choose one or the other?

Reply:
I like the command line version because it is explicit and it allows me to copy the cmake invocation which should also work without the spack build environment. On the other hand if you don't run spack install xyz first, and directly jump into spack build-env xyz, it's useful to have the environment variable so that you don't have to write the prefix path list by hand.